core/packages/contracts/contracts/BeefyClient.sol

```diff
@@ -350,7 +353,9 @@ contract BeefyClient is Ownable {
         }
 
         currentValidatorSet = nextValidatorSet;
-        nextValidatorSet.id = leaf.nextAuthoritySetID;
+        // Workaround for change in https://github.com/paritytech/polkadot/pull/6577
+        // and discussions here in https://github.com/paritytech/substrate/issues/11797
+        nextValidatorSet.id = leaf.nextAuthoritySetID + 1;
```

Comment:
I don't think this is the correct change. Detecting an authority set handover has always been tricky, and now the algorithm has changed again. 

We essentially need to find and post a (commitment, MMRLeaf) pair to the beefy client, such at commitment.validatorSet < leaf.nextAuthoritySetID and leaf.parentNumber < commitment.BlockNumber.

I'm keen to understand the problem myself, so will checkout your branch and look at the relayer to see if it can find such a pair.

I will spend some time this week looking into this.

Reply:
@vgeddes Yes, exactly the case. 

I did not find that pair as you mentioned and current status is
`commitment.validatorSet = leaf.nextAuthoritySetID` when `commitment.BlockNumber = leaf.parentNumber +1` from handover log above which is not expected as before.

I also not like this tricky workround here and thanks for checking this.

Reply:
@vgeddes I am not sure we can find that pair you mentioned after https://github.com/paritytech/polkadot/pull/6577 , 

In this PR Parity just change the order to make MMR node generated before Session change, then break point of Beefy validatorSet change from `10n` to `10n+1` (assume 10 is the session length of Rococo).  So `nextAuthoritySetID` value change in leaf data now is also as expected.

And I think the workround here is not a big issue for us, no trade off on security. Please let me know what you think about.

Reply:
@vgeddes fixed in this commit.
https://github.com/Snowfork/snowbridge/pull/780/commits/c7409e4e1099d8c3306d97d25c6b8dbcc4432e56

Reply:
Awesome! yeah looks much better. Because when `leaf.nextValidatorSetId` changes, it is expected that `leaf.nextValidatorSetLen` and `leaf.nextValidatorSetRoot` will change too.